index.js

```diff
@@ -61,10 +61,214 @@ const SCHEMA_BUILD_DATA = Joi.object().keys({
+    const isMinimized = hoek.reach(buildData, metaMinimizedReplaceVar, {
+        default: buildData.settings.slack.minimized
+    });
+
+    let message = isMinimized
+        ? // eslint-disable-next-line max-len
+            `<${pipelineLink}|${buildData.pipeline.scmRepo.name}#${buildData.jobName}> *${notificationStatus}*`
+        : // eslint-disable-next-line max-len
+            `*${notificationStatus}* ${STATUSES_MAP[notificationStatus]} <${pipelineLink}|${buildData.pipeline.scmRepo.name} ${buildData.jobName}>`;
+
+    const rootDir = hoek.reach(buildData, 'pipeline.scmRepo.rootDir', { default: false });
+
+    if (rootDir) {
+        message = `${message}\n*Source Directory:* ${rootDir}`;
+    }
+
+    const metaMessage = hoek.reach(buildData, 'build.meta.notification.slack.message', { default: false });
+    const metaVar = `build.meta.notification.slack.${buildData.jobName}.message`;
+    const buildMessage = hoek.reach(buildData, metaVar, { default: false });
+
+    // Use job specific message over generic message.
+    if (buildMessage) {
+        message = `${message}\n${buildMessage}`;
+    } else if (metaMessage) {
+        message = `${message}\n${metaMessage}`;
+    }
+
+    const attachments = isMinimized
+        ? [
+                {
+                    fallback: '',
+                    color: COLOR_MAP[notificationStatus],
+                    fields: [
+                        {
+                            title: 'Build',
+                            value: `<${buildData.buildLink}|#${buildData.build.id}>`,
+                            short: true
+                        }
+                    ]
+                }
+            ]
+        : [
+                {
+                    fallback: '',
+                    color: COLOR_MAP[notificationStatus],
+                    title: `#${buildData.build.id}`,
+                    title_link: `${buildData.buildLink}`,
+                    // eslint-disable-next-line max-len
+                    text:
+                        `${commitMessage} (<${buildData.event.commit.url}|${truncatedSha}>)` +
+                        `\n${buildData.event.causeMessage}`
+                }
+            ];
+
+    const slackMessage = {
+        message,
+        attachments
+    };
+
+    slacker(this.config.token, buildData.settings.slack.channels, slackMessage);
+}
+
+/**
+ * Handle slack messaging for job status 
+ * @method jobStatus
+ * @param  {Object}         jobData
+ * @param  {String}         jobData.status             Status
+ * @param  {Object}         jobData.pipeline           Pipeline
+ * @param  {String}         jobData.jobName            Job name
+ * @param  {String}         jobData.pipelineLink       Pipeline link
+ * @param  {String}         jobData.message            Message
+ * @param  {Object}         jobData.settings           Notification setting
+ */
+function jobStatus(jobData) {
```

Comment:
same comment here: is it possible to refactor buildStatus/jobStatus into one method? or somehow reuse more overlapping code

Reply:
anwsered in other pr